src/execution/syscall_utils.rs

```diff
@@ -94,6 +96,31 @@ pub fn felt_from_ptr(vm: &VirtualMachine, ptr: &mut Relocatable) -> Result<Felt2
     Ok(felt)
 }
 
+pub fn read_felt_array<TErr>(vm: &VirtualMachine, ptr: &mut Relocatable) -> Result<Vec<Felt252>, TErr>
```

Comment:
this should describe how it derives the array size and location (reads consecutive pointers for start/end)

and/or maybe just make it non-pub

Reply:
I think it is pretty clear what it is doing from its source.